middle_end/flambda2/from_lambda/lambda_to_flambda_primitives.ml

```diff
@@ -141,18 +141,19 @@ let convert_array_kind_for_length kind : P.Array_kind_for_length.t =
   | Float_array_opt_dynamic -> Float_array_opt_dynamic
 
 module Array_ref_kind = struct
-  (* CR mshinwell/ncourant: split into a two-level type so that
-     [Naked_floats_to_be_boxed] is separate *)
-  type t =
+  type no_float_array_opt =
     | Immediates
     | Values
-    | Naked_floats_to_be_boxed of L.locality_mode
     | Naked_floats
     | Naked_float32s
     | Naked_int32s
     | Naked_int64s
     | Naked_nativeints
     | Naked_vec128s
+
+  type t =
+    | No_float_array_opt of no_float_array_opt
```

Comment:
Could we actually use `P.Array_load_kind.t`? And likewise with `Array_set_kind`?

Reply:
On the unboxed products dev branch these are different, but it might be possible to combine them.  I will leave a CR for now.